main.py

```diff
@@ -252,6 +252,32 @@ def list(self, dpid=None):
 
         return jsonify(switch_flows)
 
+    @listen_to('kytos.flow_manager.flows.(install|delete)')
+    def event_add_flow(self, event):
+        """Install or delete flows in the switches through events.
+
+        Install or delete Flow of switches identified by dpid.
+        """
+        try:
+            dpid = event.content['dpid']
+            flow_dict = event.content['flow_dict']
+        except KeyError as error:
+            log.error("Error getting fields to install or remove "
+                      f"Flows: {error}")
+            return
+
+        if event.name == 'kytos.flow_manager.flows.install':
+            command = 'add'
+        else:
+            command = 'delete'
+
+        try:
+            switch = self.controller.get_switch_by_dpid(dpid)
+            self._install_flows(command, flow_dict, [switch])
+        except InvalidCommandError as error:
```

Comment:
Only the code that raise the exception should be inside the try/except.

```suggestion
        switch = self.controller.get_switch_by_dpid(dpid)
        try:
            self._install_flows(command, flow_dict, [switch])
        except InvalidCommandError as error:
```